x-pack/plugin/security/src/internalClusterTest/java/org/elasticsearch/xpack/security/profile/ProfileCancellationIntegTests.java

```diff
@@ -180,12 +195,20 @@ public SearchBlockPlugin(Settings settings, Path configPath) throws Exception {
         public void onIndexModule(IndexModule indexModule) {
             super.onIndexModule(indexModule);
             indexModule.addSearchOperationListener(new SearchOperationListener() {
+
+                @Override
+                public void onPreQueryPhase(SearchContext c) {
+                    logger.info("onPreQueryPhase");
+                }
```

Comment:
The blocking is done in the `onNewReaderContext` method. If for any reason, the reader context already exists before the search request, the method may not be called. So I added simple logging as the implementation of `onPreQueryPhase` to indicate that if it somehow happens.